usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateUniqueRouteLongNameRouteShortNameCombination.java

```diff
@@ -0,0 +1,78 @@
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.apache.logging.log4j.Logger;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.routes.Route;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.warning.DuplicateRouteLongNameRouteShortNameCombinationNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Use case to validate that combination of fields `route_long_name` and `route_short_name` from {@link Route} entities
+ * are unique. This use case is triggered after completing the {@code GtfsDataRepository} provided in the constructor
+ * with {@code Route} entities.
+ */
+public class ValidateUniqueRouteLongNameRouteShortNameCombination {
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+    private final Logger logger;
+
+    /**
+     * @param dataRepo   a repository storing the data of a GTFS dataset
+     * @param resultRepo a repository storing information about the validation process
+     * @param logger     a logger used to log information about the validation process
+     */
+    public ValidateUniqueRouteLongNameRouteShortNameCombination(final GtfsDataRepository dataRepo,
+                                                                final ValidationResultRepository resultRepo,
+                                                                final Logger logger) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+        this.logger = logger;
+    }
+
+    /**
+     * Use case execution method: checks unicity of combination of fields `route_long_name` and `route_short_name`
+     * in file `routes.txt`. A notice is generated and added to the {@code ValidationResultRepository} provided in the
+     * constructor each time this requirement is not satisfied.
+     */
+    public void execute() {
+        logger.info("Validating rule 'W016 - Duplicate combination od fields`route_short_name` and `route_long_name`'");
+
+        final Map<String, Route> routeByRouteLongNameRouteShortName = new HashMap<>();
+        dataRepo.getRouteAll().forEach((routeId, route) -> {
+            final String routeShortName = route.getRouteShortName();
+            final String routeLongName = route.getRouteLongName();
+            if (routeShortName != null && routeLongName != null) {
```

Comment:
I think you should actually consider `null` value combinations as well - for example - for routeShortName, routeLongName:
* `A`, `null`
* `A`, `null`

...would be a warning, as would:
* `null`, `A`
* `null`, `A`

I see this differently than the other two warning rules, which are only considering one of two fields.

So I believe you could just eliminate this line with the null check, if we are already guaranteed to have at least one of the fields being not null from previous validation. You would need to add more unit tests to verify behavior though.

Reply:
Indeed, will do. With additional comments to explain our assumptions regarding the nullability of the two fields, as you mentioned.

Reply:
However, I think that the logic here might be more complex: we would still need to check if both fields are not null since the previous validation did not exclude routes with null names from the data repository, these would still be processed and we would need a way to exclude them from the execution of this use case.

Reply:
Doesn't the spec say that you must provide a route short name or route long name - you can't leave them both empty? I would have thought we flagged this as an error early on when importing the CSV.

Reply:
Indeed, I checked our code in the builder for `Route` entities, and this requirement was not checked. So I will implement more verification in the `route` builder `.build` method to avoid this situation.

Reply:
Thanks for the heads-up